appium/lib/screen-object/accessors/element.rb

```diff
@@ -109,15 +109,14 @@ def element_visible?(direction = :down)
         driver.no_wait
         if exists?
           driver.set_wait(default_wait)
-          scroll(direction)
           true
         else
           scroll(direction)
           false
         end
-      rescue
-       scroll(direction)
-       false
+      rescue RuntimeError
+        scroll(direction)
```

Comment:
So you are now scrolling in the direction even when element is not visible? I mean if you are going to return false for element visible then what's the point of extra scroll?